cmd/signature-v4-utils.go

```diff
@@ -260,3 +262,46 @@ func signV4TrimAll(input string) string {
 	// unicode.IsSpace() internally here) to one space and return
 	return strings.Join(strings.Fields(input), " ")
 }
+
+// checkMetaHeaders will check if the metadata from header/url is equal with the one from signed headers
+func checkMetaHeaders(signedHeadersMap http.Header, r *http.Request) APIErrorCode {
+	metaHeaders := make(http.Header)
+	regex := regexp.MustCompile("^[xX]-[aA]mz-[mM]eta-.*")
+
+	for k := range signedHeadersMap {
+		if regex.MatchString(k) {
+			metaHeaders.Add(k, r.Header.Get(k))
+		}
+	}
+
+	// take values from http header
+	for k := range r.Header {
+		if regex.MatchString(k) {
+			metaHeaders.Add(k, r.Header.Get(k))
+		}
+	}
+
+	// take values from url, if no http header
+	if len(metaHeaders) == 0 {
```

Comment:
If metaheaders is `len(0)`, you didn't copy anything above, which means that no valid values are signed and any match in `r.Form` will fail.

That means we just ignore the form values and don't check them if there is a signed value. That seems a bit strange.

Reply:
if the metaHeaders is `len(0)` after header check, will check for values from URL because some sdks are sending values like this ( example below) ( aws-sdk for php as example ) and no header is needed in this case. 
`000000003930/topic/test/1684151360?x-amz-meta-name=img_jpg&x-amz-meta-extension=jpg&x-amz-meta-test=test&x-amz-meta-xxx=20&`